seed/views/v3/properties.py

```diff
@@ -45,3 +208,769 @@ def labels(self, request):
+
+        return JsonResponse({'status': 'success', 'columns': columns})
+
+    @swagger_auto_schema(
+        request_body=AutoSchemaHelper.schema_factory(
+            {
+                'selected': ['integer']
+            },
+            required=['selected'],
+            description='A list of property ids to delete')
+    )
+    @api_endpoint_class
+    @ajax_request_class
+    @has_perm_class('can_modify_data')
+    @action(detail=False, methods=['DELETE'])
+    def batch_delete(self, request):
+        """
+        Batch delete several properties
+        """
+        property_states = request.data.get('selected', [])
+        resp = PropertyState.objects.filter(pk__in=property_states).delete()
+
+        if resp[0] == 0:
+            return JsonResponse({'status': 'warning', 'message': 'No action was taken'})
+
+        return JsonResponse({'status': 'success', 'properties': resp[1]['seed.PropertyState']})
+
+    def _get_property_view(self, pk):
+        """
+        Return the property view
+
+        :param pk: id, The property view ID
+        :param cycle_pk: cycle
+        :return:
+        """
+        try:
+            property_view = PropertyView.objects.select_related(
+                'property', 'cycle', 'state'
+            ).get(
+                id=pk,
+                property__organization_id=self.request.GET['organization_id']
+            )
+            result = {
+                'status': 'success',
+                'property_view': property_view
+            }
+        except PropertyView.DoesNotExist:
+            result = {
+                'status': 'error',
+                'message': 'property view with id {} does not exist'.format(pk)
+            }
+        return result
+
+    def _get_taxlots(self, pk):
+        lot_view_pks = TaxLotProperty.objects.filter(property_view_id=pk).values_list(
+            'taxlot_view_id', flat=True)
+        lot_views = TaxLotView.objects.filter(pk__in=lot_view_pks).select_related('cycle', 'state').prefetch_related('labels')
+        lots = []
+        for lot in lot_views:
+            lots.append(TaxLotViewSerializer(lot).data)
+        return lots
+
+    def get_history(self, property_view):
+        """Return history in reverse order"""
+
+        # access the history from the property state
+        history, master = property_view.state.history()
+
+        # convert the history and master states to StateSerializers
+        master['state'] = PropertyStateSerializer(master['state_data']).data
+        del master['state_data']
+        del master['state_id']
+
+        for h in history:
+            h['state'] = PropertyStateSerializer(h['state_data']).data
+            del h['state_data']
+            del h['state_id']
+
+        return history, master
+
+    @swagger_auto_schema_org_query_param
+    @api_endpoint_class
+    @ajax_request_class
+    def retrieve(self, request, pk=None):
+        """
+        Get property details
```

Comment:
Should we let the user know to use the `property_view_id` here to avoid confusion with the `property_state_id`?

Reply:
That's a great question, @adrian-lara what are your thoughts on this? Looking at the detail views (those that use `id` in the path), they all seem to be using property view ids. Does the user of the api know/care about property states?

Reply:
From what I've seen, frontend users have no notion of inventory -Views or -States.

Are all the property endpoints using View ID? If so, I'd definitely feel OK using just `id` and **not clarifying** it's actually a View ID.